openqabot/approver.py

```diff
@@ -131,39 +131,32 @@ def is_job_marked_acceptable_for_incident(self, job_id: int, inc: int) -> bool:
             pass
         return False
 
+    def job_acceptable(self, inc: int, res) -> bool:
+        """
+        Check each job if it is acceptable for different reasons.
+
+        Keep jobs marked as acceptable for one incident by openQA comments.
+        """
+        if res["status"] == "passed":
+            return True
+        url = "{}/t{}".format(self.client.url.geturl(), res["job_id"])
+        if self.is_job_marked_acceptable_for_incident(res["job_id"], inc):
+            log.info(
+                "Ignoring failed job %s for incident %s due to openQA comment", url, inc
+            )
+            return True
+        log.info("Found failed, not-ignored job %s for incident %s", url, inc)
+        return False
+
     @lru_cache(maxsize=128)
     def get_jobs(self, job_aggr: JobAggr, api: str, inc: int) -> bool:
         results = get_json(api + str(job_aggr.id), headers=self.token)
-
-        # keep jobs explicitly marked as acceptable for this incident by openQA comments
-        for res in results:
-            ok_job = res["status"] == "passed"
-            if ok_job:
-                continue
-            if self.is_job_marked_acceptable_for_incident(res["job_id"], inc):
-                log.info(
-                    "Ignoring failed job %s/t%s for incident %s due to openQA comment",
-                    str(self.client.url.geturl()),
-                    res["job_id"],
-                    inc,
-                )
-                res["status"] = "passed"
```

Comment:
The previous code sets `res["status"]` here. Is this something we need to keep? I'm just asking because the commit starting with "Simplify …" doesn't make it sound like you have considered that.

Reply:
Yes, I have considered that explicitly. In my understanding the field is only set so that the subsequent "all" evaluates to passed so in a two-step process which I short-cut to a single loop here.

Reply:
Ah. I missed that. So your approach is definitely cleaner then